Book-eCommerce-Store/Controllers/ProductsController.cs

```diff
@@ -4,85 +4,88 @@
 using Book_eCommerce_Store.Services.ProductsService;
+using Book_eCommerce_Store.Services.ProductsService.Factory;
 using Microsoft.AspNetCore.Mvc;
 
 namespace Book_eCommerce_Store.Controllers
 {
     [ApiController]
-    [Route("api/v1/[controller]")]
+    [Route("api/[controller]")]
     public class ProductsController : ControllerBase
     {
-        private readonly IProductsService productsService;
+        private readonly IProductFactory productFactory;
 
-        public ProductsController(IProductsService productsService){
-            this.productsService = productsService;
+        public ProductsController(IProductFactory productFactory)
+        {
+            this.productFactory = productFactory;
         }
 
-        [HttpGet]
-        public async Task<ActionResult<List<GetProductDTO>>> Get()
+        [HttpGet("{productCategory}")]
+        public async Task<ActionResult<List<GetProductDTO>>> Get(ProductCategory productCategory)
         {
-            var response = await this.productsService.Get();
+            var response = await this.productFactory.GetProductsService(productCategory).Get();
             if (response.Success == true)
             {
                 response.Message = "Http Status OK";
                 return Ok(response);
             }
             return StatusCode(StatusCodes.Status500InternalServerError, response);
-            
+
         }
 
 
-        [HttpGet("{id}")]
-        public async Task<ActionResult<GetProductDTO>> GetById(int id)
+        [HttpGet("{productCategory}/{id}")]
+        public async Task<ActionResult<GetProductDTO>> GetById(ProductCategory productCategory, int id)
         {
-            var response = await this.productsService.GetById(id);
-            if (response.Success==true)
+            var response = await this.productFactory.GetProductsService(productCategory).GetById(id);
+            if (response.Success == true)
             {
                 response.Message = "Http Status OK";
                 return Ok(response);
             }
             return StatusCode(StatusCodes.Status500InternalServerError, response);
         }
 
-        [HttpPost]
-        public async Task<ActionResult<GetProductDTO>> CreateProduct(CreateProductDTO newProduct)
+        [HttpPost("{productCategory}")]
+        public async Task<ActionResult<GetProductDTO>> CreateProduct(ProductCategory productCategory, CreateProductDTO newProduct)
         {
-            var response = await this.productsService.CreateProduct(newProduct);
-            if (response.Success==true)
+            var response = await this.productFactory.GetProductsService(productCategory).Create(newProduct);
+            if (response.Success == true)
             {
                 response.Message = "Http Status OK";
                 return Ok(response);
             }
             return StatusCode(StatusCodes.Status500InternalServerError, response);
         }
 
-        [HttpPut("{id}")]
```

Comment:
Since you can do a partial update on a Product we should change from the HTTP PUT request method to the HTTP PATCH request method.